pkg/shares/reserved_bytes.go

```diff
@@ -0,0 +1,42 @@
+package shares
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+)
+
+// NewReservedBytes returns a byte slice of length
+// appconsts.CompactShareReservedBytes that contains a varint of the byteIndex
+// of the first unit that starts in a compact share. If no unit starts in the
+// compact share, ReservedBytes is [0, 0].
+func NewReservedBytes(byteIndex uint64) (reservedBytes []byte, err error) {
+	if byteIndex >= appconsts.ShareSize {
+		return []byte{}, fmt.Errorf("byte index %d must be less than share size %d", byteIndex, appconsts.ShareSize)
+	}
+	reservedBytes = make([]byte, appconsts.CompactShareReservedBytes)
+	bytesWritten := binary.PutUvarint(reservedBytes, byteIndex)
+	if bytesWritten > appconsts.CompactShareReservedBytes {
```

Comment:
> I don't think this case will ever occur, according to docs: https://pkg.go.dev/encoding/binary#PutUvarint

Agreed! Thanks for pointing out. I'll remove this conditional.

> A better check might be bytesWritten == appconsts.CompactShareReservedBytes, but I don't know if the uint64 is small, whether it will write all bytes (even the zero ones) or not.

Since we initialize the reservedBytes slice to be of length `appconsts.CompactShareReservedBytes`, even if the PutUvarint only writes a value to the first byte, the second byte will remain zero. Note this is confirmed by this [test case](https://github.com/rootulp/celestia-app/blob/8ea9c826692c675bca6d67a93a547513a2d36676/pkg/shares/reserved_bytes_test.go#L58)

> Another check might be <= 0 to check if something was written or not. But needs a bit more investigations.

Since a uint64 can't be `< 0`, I don't think this check is necessary.

Reply:
Resolve in https://github.com/celestiaorg/celestia-app/pull/850/commits/b2ce5421a00c955a402ee3cbc08a416cfa7564cc